torch/nn/utils/_expanded_weights/expanded_weights_utils.py

```diff
@@ -0,0 +1,122 @@
+        ctx: The context from the autograd.Function object. Will be used to save
+          computed state from the forward pass
+        expanded_args: Arguments to be passed to :attr:`func`. Will include arguments
+          that need to be unpacked because they are ExpandedWeights
+        num_true_outs: The number of outputs seen by the user since some functions
+          return auxillary data that is only used in the backward pass
+    '''
+    unexpanded_args = _check_and_unexpand_args(func, ctx, expanded_args)
+    output = func(*unexpanded_args)
+    return _check_and_detach_output(ctx, output, num_true_outs)
+
+def _check_and_unexpand_args(func, ctx, expanded_args):
+    ctx.args = expanded_args
+    # input must be the first argument passed
+    input = expanded_args[0]
+    if isinstance(input, ExpandedWeight):
+        raise RuntimeError("Expanded Weights do not support inputs that are also ExpandedWeights. "
+                           f"Input must be a Tensor, got {type(input).__name__} in function {func.__name__}")
+    if not isinstance(input, torch.Tensor):
+        raise RuntimeError("Expanded Weights requires a Tensor as the first input to get the batch dimension, "
+                           f"got {type(input).__name__} in function {func.__name__}")
+    if len(input.shape) == 0:
+        raise RuntimeError(f"Expanded Weights requires a batch dimension but got an input of size 0 in function {func.__name__}")
+    if input.shape[0] == 0:
+        raise RuntimeError("0 is not a valid batch size for Expanded Weights but got input tensor of "
+                           f"{input} in function {func.__name__}")
+    batch_size = input.shape[0]
+    for arg in expanded_args:
+        if isinstance(arg, ExpandedWeight) and arg.batch_size != batch_size:
+            raise RuntimeError("Expected ExpandedWeights to have batch size matching input but got "
+                               f"input batch size of {batch_size} with ExpandedWeight of batch size {arg.batch_size}")
+
+    unexpanded_args = tuple(arg.orig_weight if isinstance(arg, ExpandedWeight) else arg for arg in expanded_args)
+    return unexpanded_args
+
+def _check_and_detach_output(ctx, output, num_true_outs):
+    ctx.all_outputs = output
+
+    # separates differentiable outputs from outputs only needed for the backwards computation
+    if isinstance(output, tuple):
+        if len(output) < num_true_outs:
+            raise RuntimeError(f"Got fewer outputs ({len(output)}) than expected ({num_true_outs}). "
+                               "Issues in ExpandedWeights' autograd.Function")
+        if num_true_outs == 1:
+            output = output[0]
+        else:
+            output = output[:num_true_outs]
+    elif num_true_outs != 1:
+        raise RuntimeError(f"Got single output but expected at least {num_true_outs} outputs. "
+                           "Issues in ExpandedWeights' autograd.Function")
+    ctx.true_outputs = output
```

Comment:
Fixed--moved handling to the function level (i.e. group_norm, instance_norm, and layer_norm now unwrap the tuple of a single element) so that we could support (Tensor,) return in the future